subprojects/core/src/main/java/org/gradle/api/internal/artifacts/JavaEcosystemSupport.java

```diff
@@ -118,6 +121,65 @@ private static void configureLibraryElements(AttributesSchema attributesSchema,
         });
     }
 
+    private static void configureView(AttributesSchema attributesSchema, final ObjectFactory objectFactory) {
+        AttributeMatchingStrategy<CompileView> viewSchema = attributesSchema.attribute(CompileView.VIEW_ATTRIBUTE);
+        viewSchema.getCompatibilityRules().add(CompileViewCompatibilityRules.class);
+        viewSchema.getDisambiguationRules().add(CompileViewDisambiguationRules.class, actionConfiguration -> {
+            actionConfiguration.params(objectFactory.named(CompileView.class, CompileView.JAVA_API));
+            actionConfiguration.params(objectFactory.named(CompileView.class, CompileView.JAVA_COMPLETE));
+        });
+    }
+
+    @VisibleForTesting
+    public static class CompileViewDisambiguationRules implements AttributeDisambiguationRule<CompileView>, ReusableAction {
```

Comment:
I know you're following the existing pattern, but I think these types should:

1. Not have plural names (each is a single instance of a rule)
2. Be named after what the rule actually does, instead of the type of view it runs against
3. Have javadoc rather than inline comments to describe what they do, especially if the name remains uninformative as it presently is

Reply:
I'm going to choose to defer this for now for the sake of reducing further changes.